crates/sui-core/src/authority/authority_store_tables.rs

```diff
@@ -201,6 +201,44 @@ impl AuthorityPerpetualTables {
             .next()
             .is_none())
     }
+
+    pub fn iter_live_object_set(&self) -> LiveSetIter<'_> {
+        LiveSetIter {
+            iter: self.parent_sync.keys(),
```

Comment:
I thought we were going to get rid of `parent_sync`? Doesn't this adds a hard dependency

Reply:
We'll have to fix this when we remove parent_sync. Currently, parent_sync is both necessary and sufficient for enumerating live objects (because of tombstones) so there is no way to implement this without consulting parent sync.

When we remove parent sync, we'll have to put tombstones directly in the objects table, at which point we can change this code to iterate over the objects table.

Reply:
Do tombstones need to live forever?

Reply:
they need to live until the object they mark dead has been pruned. currently parent_sync is not pruned - but tombstones could be pruned from it if they refer to objects that do not exist in the objects table.

when we consolidate parent_sync into the objects table (probably after https://github.com/MystenLabs/sui/pull/7040) then tombstone objects can be pruned if all preceding objects have been pruned.